### PR TITLE
Continuous deployment to Heroku using Travis CI

### PR DESCRIPTION
**Pull request by chrisjmit**
Hi all, Travis CI integration is now working with tests passing. We can't deploy to Heroku unless at master branch so not been able to test final deploy yet. 
Travis will approve the content of merged pull requests onto the Heroku server, making approved content live for each merge going forward.
Please could you review and ship as appropriate, as soon as you can? There's a couple of settings that can be really tricky for this and can prevent pull requests from happening.
Apologies for the faffs - it's really not been clear.
Thanks,
Jess & Chris

### CODE REVIEW

**Comment by sophieklm** [comment]
:shipit:

**Review by milesillsley — CHANGES_REQUESTED** [review]
- remove one of the scripts in .travis.yml (line 5)
- there is a lonely # in Gemfile (line 9)
Code looks great, though Im not really sure how to test it or run the specs

**Comment by chrisjmit** [comment]
test suggestion: try logging in to travis using github credentials, and from profile page adding the repo via organizations/makers academy. the branch test log should be for continuous-deployment

**Comment by JessicaBarclay** [comment]
@milesillsley Thanks - changes made

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,5 +1,6 @@
+<% format_url = Classes::FormatUrl.new %>
 <% @posts.each do |post| %>
-  <p><%= post.message %></p>
+  <p><%= format_url.return_formatted_url(post.message).html_safe %></p>
```
Comment (sblausten) [inline app/views/posts/index.html.erb]: Can't figure out what this is doing. What's the change for?

**Comment by sblausten** [comment]
@chrisjmit seems to be lots of code here adding new feature to posts that renders hyperlinks correctly in post display. Seems to be separate from Travis issue? 

Would be good to take that out and put in a new branch.

File: app/models/classes/format_url.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Classes::FormatUrl
+
+  def return_formatted_url(post)
```
Comment (Robert-G-J) [inline app/models/classes/format_url.rb]: How does this relate to the Travis deployment? Looks like a different issue 😕

Comment (chrisjmit) [reply app/models/classes/format_url.rb]: this seems to be have been git added in via a code review of another branch. removed.

**Comment by chrisjmit** [comment]
@sblausten this seems to be have been git added in via a code review of another branch. removed.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...

+  - bundle exec bin/rake db:migrate

+  - bundle exec rspec spec
+
+deploy:
+  provider: heroku
+  app: ace-book
+  api_key:
+    secure: "Zhq+bAklzTvNgYeVKtA5mi3GmTYqEzCWCGclgOuK20D23Yi1iPkbcH09ricLABlk1Y7qFB8jGtbplrPbGrQQdhUW9sQ9OGLRlA/vJnMhU4IVK/ZM1EJCFZPZPv5Zbg4s8L2r0Gz7uIEOX05TYFcREZ5iice4Mq89rX/SgOVPswh3KCFq0BfxaLq111WNAJtApH9amhuJbOUnRB7zMu38kpJcUqMsCFK3jeYBMbp4GxTyg+KK9MtiHS1tOrtXttx5EGaRyzBVi3JgFBo7gTaL3d03FJ6S5/3RFtDol3cUMSIfUVJkFlnzoedmyIgRq+PeDFIUZ17U6ZW8H09Rz9DjXBrpbzIGTEH92Xij7Fqd/BB2PE0ygSFwMcAxGvJHE5WJqNe6T5BhkATxGPXlYwRWMg3szZJdvlsu68xn6xAPdm2xH1DvdL1/ptcWltQFThjp3iy494sODbUInnjHPrj6zr1GsYlWnHigjcqi0IoJFjZRe55gM31cAadFYSBtm9wVoZHzyEje4MmjUwSUZMlaUVkv28eN5rdt8dbZYznu6IPUEY1nTFFipyiDq9WwidyziS4D6Flm5pbcNQBHyg+YwC1fzu3mcMSBnJtzqA60+IYao6iI+uI1eMX3+gSNdUVfEug8O6nuBxD5nCyojMlqmczai/s5oRJeTMT3PvITOV4="
```
Comment (dbugsy) [inline .travis.yml]: Woah this should not be in git. What is that the api key for? Heroku? Are you 100% sure you need this in the yml file? I would suggest immediately changing this key in heroku, as it is now compromised.

**Review by dbugsy — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 3 above)
Comment (chrisjmit) [reply .travis.yml]: @ddosque this isn't the api-key, but the travis gem encrypted api key. we installed the travis gem and ran travis encrypt (api-key-hash) --add deploy.api_key to generate this change. we've redone this and also done this manually to the same result. we'll look into another way of doing this for heroku if we're off the mark.

Comment (chrisjmit) [reply .travis.yml]: Reference:
https://docs.travis-ci.com/user/deployment/heroku/

**Review by dbugsy — APPROVED** [review]

**Comment by dbugsy** [comment]
Merge this anyway, even with a failing build - it will prevent further red tests going into production. There is a ticket to fix the time issues.

**Pull request merged**